File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/java/field_assign_to_array_item.java
class Foo {
    String field;
    String other;
}
class A {
    int main() {
        Foo f11 = new Foo();
        String[] f22 = new String[10];
        f22[1] = null;
        f11.field = f22[1];
        return f11.field.length(); // should be detected here
    }
}

File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/field_assign_to_array_item.cpp
class Foo {
public:
    int *field;
    int *other;
};

int main() {
    Foo f11;
    int* f22[10];
    f22[1] = 0;
    f11.field = f22[1];
    return *f11.field; // should be detected here
}
